format/src/main/scala/org/scalafmt/internal/TreePrinter.scala

```diff
@@ -527,11 +545,31 @@ object TreePrinter {
           case t: Pat.Interpolate =>
             dInterpolate(t.prefix, t.parts, t.args)
           case t: Pat.Typed =>
-            val drhs = t.rhs match {
-              case l: Lit => Literal.wrap(l)
-              case rhs => RefineTyp.wrap(rhs)
+            print(t.lhs) + `:` + space + RefineTyp.wrap(t.rhs)
+          case t: Pat.Xml =>
```

Comment:
The logic here is quite large compared to the other cases, would be nice to extract into a `d...` function.